crates/symbolicator/src/services/symbolication/module_lookup.rs

```diff
@@ -83,7 +83,11 @@ impl ModuleLookup {
         modules.sort_by_key(|entry| entry.object_info.raw.image_addr.0);
 
         // Ignore the name `dedup_by`, I just want to iterate over consecutive items and update
-        // some.
+        // some. That case would rather be served by a `windows_mut` iterator, but that does not
+        // exist (and probably never will because it would yield overlapping mutable pairs,
+        // violating exclusive mutable borrow rules)
+        // Also NOTE that `dedup_by` is defined like so (entry1/2 are swapped):
+        // > The elements are passed in opposite order from their order in the slice
```

Comment:
So I guess we can continue the conversation of https://github.com/getsentry/symbolicator/pull/685#discussion_r811302464 here.  I'm still convinced this is a poor choice:

- It is API abuse, you need a visitor but you are using something that's a visitor as a side-effect.
- The justification is now 6 lines long, that says it all really.
- This now talks about `windows_mut`, but this is an entirely different API as it generates a new iterator and is not a visitor (and therefore can't exist as explained).  Mixing visitors and iterators is another example of how this is confusing.
- It is complex to reason about as demonstrated by all the comments, code and otherwise.

So really I would urge again to not use this and do it the simple boring way rather than the clever way.

Reply:
I agree with this.